Source/WebSocket_Android.cpp

```diff
@@ -0,0 +1,89 @@
+#include "WebSocket_Base.h"
+#include <AndroidExtensions/Globals.h>
+#include <AndroidExtensions/JavaWrappers.h>
+
+using namespace android::global;
+using namespace android::net;
+using namespace java::lang;
+using namespace java::io;
+using namespace java::net;
+using namespace java::websocket;
+
+namespace UrlLib
+{
+    class WebSocket::WSImpl : public WSImplBase
+    {
+    public:
+        ~WSImpl()
+        {
+            delete m_webSocket;
+        }
+
+        void Open(std::string url,
+            std::function<void(void)> onopen,
+            std::function<void(void)> onclose,
+            std::function<void(std::string)> onmessage,
+            std::function<void(void)> onerror)
+        {
+            open_callback = onopen;
+            close_callback = onclose;
+            message_callback = onmessage;
+            error_callback = onerror;
+
+            m_url = url;
+            m_readyState = ReadyState::Connecting;
+            m_webSocket = new WebSocketClient(url, open_callback_stored, close_callback_stored, message_callback, error_callback);
+        }
+
+        void Send(std::string message)
+        {
+            if (!m_webSocket)
+            {
+                throw std::runtime_error{"WebSocket is not initialized"};
+            }
```

Comment:
If we want to do these type of checks, it should be in all of the public functions.